ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,420 @@
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union
+
+import torch
+
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+def iou(y: torch.Tensor, y_pred: torch.Tensor, crowd: List) -> torch.Tensor:
+    m = len(y_pred)
+    n = len(y)
+
+    ious = torch.zeros(m, n)
+
+    for g in range(n):
+        Y = y[g].tolist()
+        ya = Y[2] * Y[3]
+        iscrowd = crowd[g]
+        for d in range(m):
+            P = y_pred[d].tolist()
+            pa = P[2] * P[3]
+            ious[m - d - 1, g] = 0
```

Comment:
Why the array is filled in reverse order ?

Reply:
Because of confidence sorting used in self._eval_images and self._compute_iou.